opus/application/static_media/js/browse.js

```diff
@@ -674,6 +674,11 @@ var o_browse = {
 
     // find the first displayed observation index & id in the upper left corner
     updateSliderHandle: function(browserResized=false) {
+        // Only update the slider & obSnum in infiniteScroll instances when the user
```

Comment:
What is the point of this new function? Doesn't it do exactly the same thing as getViewTab below?

Reply:
This is directly reading from opus.prefs.view and it will have possible returned value equal to "search", "browse", "cart", and "detail". It will also let us know where users are at right now, for example, it will return "search" if the user is at search page. getViewTab only returns "#cart" or "#browse", and if the user is at search page, the function still returns "#browse".  The reason of using this function is to make sure updateSliderHandle only gets called when we select browse tab (or we are at browse tab). This is to avoid updateSliderHandle being called from adjustBrowseHeight when mutationObserver detects a tab changes. For example, when we are in browse view and click search nav-link, mutationObserver will detect a change and call adjustBrowseHeight even though we are getting out of browse view. By adding the check here, updateSliderHandle will not be called when we are exiting out the browse tab, so the startObs stored in the infiniteScroll instances and startobs in url will not get messed up (because the calculation in updateSliderHandle is based on the existing elements in browse tab, variables like topBoxBoundary will get messed up in above example).

Reply:
Ah, I see. I think we might want to rename "getViewTab" to "getViewBrowseCartTab" or something to avoid confusion.